components/sync/driver/brave_sync_service_impl.cc

```diff
@@ -142,4 +142,33 @@ void BraveSyncServiceImpl::ResumeDeviceObserver() {
   sync_service_impl_delegate_->ResumeDeviceObserver();
 }
 
+void BraveSyncServiceImpl::OnEngineInitialized(
+    const WeakHandle<DataTypeDebugInfoListener>& debug_info_listener,
+    bool success,
+    bool is_first_time_sync_configure) {
+  SyncServiceImpl::OnEngineInitialized(debug_info_listener, success,
+                                       is_first_time_sync_configure);
+  if (!IsEngineInitialized()) {
+    return;
+  }
+
+  syncer::SyncUserSettings* sync_user_settings = GetUserSettings();
+  if (!sync_user_settings->IsFirstSetupComplete()) {
+    return;
+  }
+
+  bool failed_to_decrypt = false;
+  std::string passphrase = brave_sync_prefs_.GetSeed(&failed_to_decrypt);
+  if (passphrase.empty()) {
```

Comment:
we should also check failed_to_decrypt here

Reply:
I have added `DCHECK` at https://github.com/brave/brave-core/compare/854d061546e5daf9b3a90267e1c752def2c0375f..ee92ac04ab82e041dd5170bd2e5b0512db57610c .

I admit, I can't do more. If `passphrase` is empty, and `failed_to_decrypt` is `true` - this means that we are on macOS and we could not get access to the keychain  => we cannot get plain `passphrase` and we cannot pass it into `sync_user_settings->SetDecryptionPassphrase` even if we understand that we need to pass it.

Reply:
actually I was referring to `passphrase.empty() || failed_to_decrypt`

Reply:
but if `failed_to_decrypt` is true, then `passphrase` is always empty

Reply:
ok, then DCHECK looks good